### PR TITLE
Upgrade Equipment page to grid display

### PR DESCRIPTION
**Pull request by jazeee**
Remove Weight table, and reorganize equipment table to use maximum page width.
Separate out the overview styles from the equipment styles.
Clean up some overview styles.
Consolidate table macros to be general use.
Fix a misalignment in spells grid due to miscounted column count. (now programmatic)

## Screenies

| Old | new |
| ---- | ---- |
| image | image |
| image | image |
| image | image |

### CODE REVIEW

File: ui/basicTab/basicTab.html [hunk 1]
```diff
@@ -148,36 +148,5 @@
         <input name="attr_featCP" class="featCP" value="" type="number" />
       </fieldset>
     </div>
-
-    &emsp; &emsp;
-    <!-- weight penalty table -->
-    <div class="flex-stack">
-      <div class="center wideish bold">Weight Penalties</div>
```
Comment (jazeee) [inline ui/basicTab/basicTab.html:155]: Weight penalties are now redundant with the header.
We can remove this in favor of opening up more space for the tab content.

File: ui/equipment/equipment.scss [hunk 2]
```diff
@@ -1,110 +1,77 @@
 /* -------- Equipment --------- */
-
-/* overview section */
-
-.weapons-overview div[data-groupname] span {
```
Comment (jazeee) [inline ui/equipment/equipment.scss:5]: Managing styles across tabs is not really easy.
Instead, we can use macros and variables in `scss` if we need some consistencies.
This section is now moved to the `overviewTab.scss` and restricted to `.weapons-overview` class parent.

File: ui/global/global.scss [hunk 3]
```diff
@@ -54,7 +55,7 @@
   width: 60px;
 }
 
-.wide.wide.wide {
+.wide.wide.wide.wide {
```
Comment (jazeee) [inline ui/global/global.scss:58]: Working on this pain point.

File: ui/global/global.scss [hunk 4]
```diff
@@ -66,8 +67,13 @@
   line-height: 1;
 }
 
+
 .spacer {
-  height: 20px;
+  height: 16px;
```
Comment (jazeee) [inline ui/global/global.scss:72]: Moving to multiples of 8 sizing.

File: ui/overviewTab/overviewTab.scss [hunk 5]
```diff
@@ -34,3 +34,63 @@
 .advantage-overview>div.conditional-something {
   line-height: 1;
 }
+
+.weapons-overview {
```
Comment (jazeee) [inline ui/overviewTab/overviewTab.scss:38]: Moved from `equipment` scss file.

File: ui/spells/spells.html [hunk 6]
```diff
@@ -44,7 +44,7 @@
     <input name="attr_arcane_total_TP" type="hidden" value="0" />
     <input name="attr_arcane_total_CP" type="hidden" value="0" />
     <input name="attr_arcane_total_single_CP" type="hidden" value="0" />
-    <div class="arcane spells-header-grid">
+    <div class="arcane epic-table-header-grid">
```
Comment (jazeee) [inline ui/spells/spells.html:47]: I consolidated into a general usage `epic-table` reusable macro style.

File: ui/spells/spells.scss [hunk 7]
```diff
@@ -130,110 +132,50 @@
...
+    0.5fr,
     // Exper.
-    minmax(0, 1fr)
+    1fr,
     // IQ ±
-    minmax(0, 0.5fr)
+    0.5fr,
     // Base
-    minmax(0, 1fr);
+    1fr,
```
Comment (jazeee) [inline ui/spells/spells.scss]: All of these `minmax`'s are auto applied now.

File: ui/spells/spells.scss [hunk 8]
```diff
@@ -130,110 +132,50 @@
...
+  );
 
-  @include spells-grid('arcane', $arcane-grid-template, 18);
+  @include epic-table-grid('arcane', $arcane-grid-template);
```
Comment (jazeee) [inline ui/spells/spells.scss:155]: I eliminated the need to specify the length of the grid template array. This 18 was actually incorrect. It should have been 17.
It caused a misalignment in the UI:

| Old | new |
| --- | --- |
| image | image |

File: ui/components/gridTable/gridTable.scss [hunk 9]
```diff
@@ -0,0 +1,93 @@
...
+
+@function getResponsiveGridWidths($gridWidths...) {
+  $result: ();
+  @each $width in $gridWidths {
+    $result: list.append($result, minmax(0, $width))
+  }
+  @return $result;
+}
+
+@mixin grid-table-tp-cp-columns {
```
Comment (jazeee) [inline ui/components/gridTable/gridTable.scss:74]: Moved this here to centralize cp/tp styles within a grid table.

**Review by johnlamping — APPROVED** [review]

**Pull request merged**